plugins/GSdx/GSRendererHW.cpp

```diff
@@ -881,6 +881,23 @@ void GSRendererHW::Draw()
 		m_texture_shuffle = (GSLocalMemory::m_psm[context->FRAME.PSM].bpp == 16) && (tex_psm.bpp == 16)
 			&& draw_sprite_tex && tex->m_32_bits_fmt;
 
+		size_t count = m_vertex.next;
+		GSVertex* v = &m_vertex.buff[0];
+
+		// Shadow_of_memories_Shadow_Flickering (Okami mustn't call this code)
+		if (m_texture_shuffle && count < 3 && PRIM->FST && (m_context->FRAME.FBMSK == 0)) {
```

Comment:
2 small nitpicks
* Replace `count` by `m_vertex.next` (remove count declaration).
* Move `GSVertex* v = &m_vertex.buff[0];` below i.e just after the `GL_INS("WARNING: Possible misdetection of a texture shuffle effect");`